localstack/services/stepfunctions/asl/component/state/state_execution/state_task/lambda_eval_utils.py

```diff
@@ -0,0 +1,55 @@
+import json
+from json import JSONDecodeError
+from typing import Any, Final, Optional
+
+from localstack.aws.api.lambda_ import InvocationResponse
+from localstack.services.stepfunctions.asl.eval.environment import Environment
+from localstack.services.stepfunctions.asl.utils.encoding import to_json_str
+from localstack.utils.aws.aws_stack import connect_to_service
+from localstack.utils.collections import select_from_typed_dict
+from localstack.utils.run import to_str
+from localstack.utils.strings import to_bytes
+
+
+class LambdaFunctionErrorException(Exception):
+    function_error: Final[Optional[str]]
+    payload: Final[str]
+
+    def __init__(self, function_error: Optional[str], payload: str):
+        self.function_error = function_error
+        self.payload = payload
+
+
+def exec_lambda_function(env: Environment, parameters: dict) -> None:
+    lambda_client = connect_to_service("lambda")
```

Comment:
```suggestion
    lambda_client = connect_to().awslambda
```
this should be replaced with our new client factory

Reply:
please also change this anywhere else where applicable